### PR TITLE
Add device tests for custom measurement processes

### PR DESCRIPTION
**Pull request by AlbertMitjans**
Test the execution of `SampleMeasurement`, `StateMeasurement` and `MeasurementTransform` measurement processes in `devices/tests`.

These tests can be executed using `lightning.qubit` and `lightning.kokkos` devices:
```
$ pl-device-test --device lightning.qubit
====================== 388 passed, 16 skipped, 224 warnings in 3.44s ========================
```
```
$ pl-device-test --device lightning.kokkos
====================== 374 passed, 30 skipped, 224 warnings in 3.36s ========================
```

### CODE REVIEW

File: pennylane/devices/tests/test_measurements.py [hunk 1]
```diff
@@ -1476,3 +1477,175 @@ def circuit(basis_state):
...
+            circuit()
+
+    def test_method_overriden_by_device(self, device):
+        """Test that the device can override a measurement process."""
+        dev = device(2)
+
+        class MyMeasurement(StateMeasurement):
+            """Dummy state measurement."""
+
+            method_name = "test_method"
```
[the remark this answers is not in the document]
Comment (josh146) [reply pennylane/devices/tests/test_measurements.py:1600]: Could we simply use the class name here? E.g.,

```python
@property
def name(self):
    return self.__name__
```
or alternatively

```python
@classmethod
def name(cls):
    return cls.__name__
```

This way there would be no redundancy --- for a developer/user using this API, the name of the measurement class is the name the device can override.

**Review by albi3ro — COMMENTED** [review]
I'll approve once i see it updated with master. If its causing problems with plugins failing during the release process, we can make the tests optional. But for now, this will show us feature coverage through the plugins, which will be really helpful.

File: pennylane/devices/tests/test_measurements.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/devices/tests/test_measurements.py:1600]: @albi3ro any device that wants to override the `MyMeasurement` would need to define a `test_method` method.

[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/devices/tests/test_measurements.py:1600]: I believe the main problem is to find a way to relate a MP class with a method defined in the `Device`. 

Another solution might be adding a dictionary in the device class mapping a MP to an overriden method name:
```python
import pennylane as qml
from pennylane.measurements import CountsMP

class CustomDevice:
    custom_measurements = {CountsMP: "counts"}

    def statistics(self, m):
        if type(m) in self.custom_measurements:
            return getattr(self, self.custom_measurements[type(m)])(m)

    def counts(self, m):
        return m.__class__.__name__

dev = CustomDevice()
print(dev.statistics(m=CountsMP(obs=qml.PauliX(0))))
```
```
CountsMP
```
This way we have the mapping information and the overriden method in the same place.

[the remark this answers is not in the document]
Comment (AlbertMitjans) [reply pennylane/devices/tests/test_measurements.py:1600]: @josh146 let me know what you think and if you have a better idea! 😄

**Review by albi3ro — APPROVED** [review]
Approved. Thanks!

**Pull request merged**